Mathlib/Data/Vector.lean

```diff
@@ -0,0 +1,257 @@
+/-
+Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura
+-/
+import Mathlib.Mathport.Rename
+import Std.Data.List.Basic
+import Std.Data.List.Lemmas
+import Mathlib.Init.Data.List.Basic
+import Mathlib.Init.Data.List.Lemmas
+
+/-!
+The type `Vector` represents lists with fixed length.
+-/
+
+universe u v w
+/-- `Vector α n` is the type of lists of length `n` with elements of type `α`. -/
+def Vector (α : Type u) (n : ℕ) :=
+  { l : List α // l.length = n }
+#align vector Vector
+
+namespace Vector
+
+variable {α : Type u} {β : Type v} {φ : Type w}
+
+variable {n : ℕ}
+
+instance [DecidableEq α] : DecidableEq (Vector α n) := by
+  unfold Vector
+  infer_instance
```

Comment:
Do we have `@[derive DecidableEq]` in Mathlib4 yet? If so then this could just be put on the `Vector` definition.